public/app/features/alerting/unified/components/notification-policies/Policy.tsx

```diff
@@ -116,41 +110,338 @@ const Policy: FC<PolicyComponentProps> = ({
+                contactPoint={contactPoint}
+                groupBy={groupBy}
+                muteTimings={muteTimings}
+                timingOptions={timingOptions}
+                inheritedProperties={inheritedProperties}
+                alertManagerSourceName={alertManagerSourceName}
+                receivers={receivers}
+                matchingAlertGroups={matchingAlertGroups}
+                matchers={matchers}
+                isDefaultPolicy={isDefaultPolicy}
+                onShowAlertInstances={onShowAlertInstances}
+              />
+            </Stack>
+          </div>
+        </div>
+        <div className={styles.childPolicies}>
+          {renderChildPolicies && (
+            <>
+              {childPolicies.map((child) => {
+                const childInheritedProperties = getInheritedProperties(currentRoute, child, inheritedProperties);
+                // This child is autogenerated if it's the autogenerated root or if it's a child of an autogenerated policy.
+                const isThisChildAutoGenerated = isAutoGeneratedRootAndSimplifiedEnabled(child) || isAutoGenerated;
+                /* pass the "readOnly" prop from the parent, because for any child policy , if its parent it's not editable, 
+                then the child policy should not be editable either */
+                const isThisChildReadOnly = readOnly || provisioned || isAutoGenerated;
+
+                return (
+                  <Policy
+                    key={child.id}
+                    routeTree={routeTree}
+                    currentRoute={child}
+                    receivers={receivers}
+                    contactPointsState={contactPointsState}
+                    readOnly={isThisChildReadOnly}
+                    inheritedProperties={childInheritedProperties}
+                    onAddPolicy={onAddPolicy}
+                    onEditPolicy={onEditPolicy}
+                    onDeletePolicy={onDeletePolicy}
+                    onShowAlertInstances={onShowAlertInstances}
+                    alertManagerSourceName={alertManagerSourceName}
+                    alertGroups={alertGroups}
+                    routesMatchingFilters={routesMatchingFilters}
+                    matchingInstancesPreview={matchingInstancesPreview}
+                    isAutoGenerated={isThisChildAutoGenerated}
+                  />
+                );
+              })}
+            </>
+          )}
+        </div>
+        {showExportDrawer && <GrafanaPoliciesExporter onClose={toggleShowExportDrawer} />}
+      </Stack>
+    </>
+  );
+};
+/**
+ * This function returns if the policy should be collapsible or not.
+ * Add here conditions for policies that should be collapsible.
+ */
+function useShoudPolicyBeCollapsible(route: RouteWithID): boolean {
+  const [isSupportedToSeeAutogeneratedChunk, isAllowedToSeeAutogeneratedChunk] = useAlertmanagerAbility(
+    AlertmanagerAction.ViewAutogeneratedPolicyTree
+  );
+  return (
```

Comment:
We might want to take in to account that the policy doesn't have any children to render (ie. we don't allow collapsible for leaf nodes), otherwise users might think the uncollapse feature is broken :)

Reply:
good catch, updating the code to not render as collapsible in this case 👍